src/main/java/calculator/service/OperationFactory.java

```diff
@@ -0,0 +1,21 @@
+package calculator.service;
+
+import calculator.domain.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationFactory {
+    private static final Map<String, Operation> operations = new HashMap<>();
+
+    static {
+        operations.put("+", new Addition());
+        operations.put("-", new Subtraction());
+        operations.put("*", new Multiplication());
+        operations.put("/", new Division());
+    }
+
+    public static Operation getOperation(String operator) {
+        return operations.get(operator);
+    }
```

Comment:
유효하지 않은 연산자가 들어올 때 예외를 던지면
외부에서 null 처리를 하지 않아도 되겠네요

null인지 아닌지 알 수 없을 때 Optional 사용을 적극 추천드립니다.

```suggestion
    public static Operation getOperation(String operator) {
        return Optional.ofNullable(operations.get(operator))
                .orElseThrow(() -> 예외);
    }
```